core/src/window_service.rs

```diff
@@ -77,26 +81,28 @@ where
     let now = Instant::now();
     inc_new_counter_debug!("streamer-recv_window-recv", packets.packets.len());
 
-    let mut shreds = vec![];
-    let mut discards = vec![];
-    for (i, packet) in packets.packets.iter_mut().enumerate() {
-        if let Ok(s) = bincode::deserialize(&packet.data) {
-            let shred: Shred = s;
-            if shred_filter(&shred, &packet.data) {
-                packet.meta.slot = shred.slot();
-                packet.meta.seed = shred.seed();
-                shreds.push(shred);
-            } else {
-                discards.push(i);
-            }
-        } else {
-            discards.push(i);
-        }
-    }
-
-    for i in discards.into_iter().rev() {
-        packets.packets.remove(i);
-    }
+    let (shreds, packets): (Vec<_>, Vec<_>) = thread_pool.install(|| {
+        packets
```

Comment:
Do you think we should have a max batch size for the packets.packets in the try_recv above?

Reply:
Yeah when we intro the gpu handoff then we can split the pathways based on items in the batch.